lib/ignored-paths.js

```diff
@@ -157,6 +181,25 @@ class IgnoredPaths {
                 this.baseDir = path.dirname(path.resolve(options.cwd, ignorePath));
                 addIgnoreFile(this.ig.custom, ignorePath);
                 addIgnoreFile(this.ig.default, ignorePath);
+            } else {
+                try {
+
+                    // if the ignoreFile does not exist, check package.json for eslintIgnore
+                    const packageJSONPath = findPackageJSONFile(options.cwd);
+
+                    if (packageJSONPath) {
+                        const configOptions = config.load(packageJSONPath, configContext);
```

Comment:
I don't think it's necessary to load the file as a config. You can just call read/parse the file as JSON to check whether it has an `eslintIgnore` key.